src/state/Maps/selectors.js

```diff
@@ -1,1181 +1,501 @@
-				} else {
-					return false;
-				}
-			});
-			if (filtered && filtered.length) {
-				return filtered[0]; //TODO what if more maps are selected based on filter
-			} else {
-				return null;
-			}
-		} else {
-			return null;
-		}
-	}
+const getBackgroundLayerStateByMapKey = createSelector(
+    [
+        getMapBackgroundLayerStateByMapKey,
+        getMapSetBackgroundLayerStateByMapKey,
+    ],
+    (mapBackgroundLayer, setBackgroundLayer) => {
+        return mapBackgroundLayer || setBackgroundLayer || null;
+    }
 );
 
 /**
  * @param state {Object}
  * @param mapKey {string}
- */
-const getNavigator_deprecated = createSelector(
-	[getMapByKey,
-		getMapSetByMapKey],
-	(map, set) => {
-		if (map) {
-			if (set) {
-				let mapNavigator = map.data && map.data.worldWindNavigator;
-				let mapSetNavigator = set.data && set.data.worldWindNavigator;
-				let navigator = {...mapSetNavigator, ...mapNavigator};
-				return !_.isEmpty(navigator) ? navigator : null;
-			} else {
-				let navigator = map.data && map.data.worldWindNavigator;
-				return navigator && !_.isEmpty(navigator) ? navigator : null;
-			}
-		} else {
-			return null;
-		}
-	}
-);
-
-const getMapSetNavigatorRange_deprecated = createSelector(
-	[
-		getMapSetByKey
-	],
-	(set) => {
-		if (set) {
-			return set.data && set.data.worldWindNavigator && set.data.worldWindNavigator.range;
-		} else {
-			return null;
-		}
-	}
-);
-
-
-
-/**
- * Collect and prepare data for map component
- *
- * @param state {Object}
- * @param layers {Array} Collection of layers data. Each object in collection contains filter property (it is used for selecting of relations) and data property (which contains data about layer from map state - e.g. key).
- */
-const getLayers_deprecated = createSelector(
-	[
-		SpatialDataSourcesSelectors.getFilteredGroupedByLayerKey,
-		AttributeDataSelectors.getFilteredGroupedByLayerKey,
-		AppSelectors.getCompleteLocalConfiguration,
-		(state, layers) => (layers)
-	],
-	/**
-	 * @param groupedSources {null | Object} Data sources grouped by layer key
-	 * @param layers {null | Array}
-	 * @return {null | Array} Collection of layers data for map component
-	 */
-	(groupedSpatialSources, groupedAttributeSources,localConfig, layers) => {
-		// FIXME - more complex
-		if (groupedSpatialSources && groupedAttributeSources && layers) {
-			return layers.map((layer) => getLayerConfiguration(localConfig, layer, groupedSpatialSources[layer.data.key], groupedAttributeSources[layer.data.key]));
-		} else {
-			return null;
-		}
-	}
-);
+ * @return {Object} Merged mapSetState with metadataModifiers and filterByActive.
+ */
+const getMapSetLayersStateByMapKeyWithModifiers = createCachedSelector(
+    [
+        getMapSetLayersStateByMapKey,
+        getMapSetMetadataModifiersByMapKey,
+        getMapSetFilterByActiveByMapKey
+    ],
+    (setLayers, metadataModifiers, mapFilterByActive) => {
+        if (setLayers?.length) {
+            setLayers = setLayers.map(layer => {
```

Comment:
Merging of metadata is same like in getMapLayersStateByMapKeyWithModifiers. Can you make some helper for it?